hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-applicationhistoryservice/src/main/java/org/apache/hadoop/yarn/server/applicationhistoryservice/ApplicationHistoryManagerOnTimelineStore.java

```diff
@@ -498,17 +498,19 @@ private ApplicationReportExt generateApplicationReport(TimelineEntity entity,
       if (app.appReport.getCurrentApplicationAttemptId() != null) {
         ApplicationAttemptReport appAttempt =
             getApplicationAttempt(app.appReport.getCurrentApplicationAttemptId());
-        if (appAttempt != null) {
```

Comment:
I don't understand why the if protection has been removed.
Don't we introduce the risk of a NullPointerException here?